meson_init.sh

```diff
@@ -65,25 +65,48 @@ FLAGS_specified_toolchain_file=false
 FLAGS_coverage=false
 ARG_toolchain_file="${TOOLCHAIN_PATH}/meson-riscv32-unknown-elf-clang.txt"
 ARG_tock_dir=""
-while getopts 'r?:f?:K?:A?:T:t:c?' flag; do
+# `getopts` usage
+# - The initial colon in the optstring is to suppress the default error
+#   handling.
+# - The remaining options are specified in alphabetical order, and the case
+#   statement should match this order.
+# - Only options that take an argument should have a following colon.
+# - The case statement contains two additional cases:
+#   - when `$flag` = `?`, this is an unexpected option.
+#   - when `$flag` = `:`, this is the case that a flag which requires an
+#     argument is not provided one. In both cases, `$OPTARG` contains the
+#     relevant parsed option.
+# - After option parsing is finished, we `shift` by `$OPTIND - 1` so that the
+#   remaining (unprocessed) arguments are in `$@` (and $1, $2, $3 etc.).
+while getopts ':AcfKrt:T:' flag; do
   case "${flag}" in
-    f) FLAGS_force=true;;
-    r) FLAGS_reconfigure="--reconfigure";;
     A) FLAGS_assert=true;;
+    c) FLAGS_coverage=true;;
+    f) FLAGS_force=true;;
     K) FLAGS_keep_includes=true;;
+    r) FLAGS_reconfigure="--reconfigure";;
     t) FLAGS_specified_toolchain_file=true
        ARG_toolchain_file="${OPTARG}";;
     T) ARG_tock_dir="${OPTARG}";;
-    c) FLAGS_coverage=true;;
-    ?) usage && exit 1;;
-    :) usage
-       error "${OPTARG} requires a path argument"
-       ;;
-    *) usage
-       error "Unexpected option ${flag}"
```

Comment:
I'd consider keeping a `*` case here, with an "Oh, no! Internal error!" message, to catch silly errors where the getopts string gets out of sync with the case statement.

Reply:
Done.